### DISCUSSION TITLE
Help setup port mirroring in virtual environment

### DISCUSSION THREAD

**Discussion by lonelyadmin** [Security-Onion-Solutions/securityonion#5388] [Unsupported Versions] [answered]
I'm having a helluva time setting up port mirroring with vsphere 7.0. I have Multiple hosts, vDS, and vDPG. Each vDPG corresponds to a vlan. Each ESX host has a dedicated sensor pinned to it with an interface that is connected to a vDS.

How to I setup port mirroring on each vDS so that it acts as a SPAN (er destination?) port for that virtual switch (and all VLANs/DPGs) on that vDS for all traffic on that host?

I'm assuming I need to use a port mirroring session of type "Distributed Port Mirroring" on the vDS...but I get lost when I'm selecting sources as it wants me to select interfaces. I'm thinking that I would want to capture traffic on any VLAN on that particular host, not just interfaces as they exist at that point in time.

E.g. vDPG named "Test" that uses VLAN 50 is a vDPG on vDS "Data" all of which are available across all ESX hosts, but I only want to capture vlan 50 traffic for the host this particular sensor resides on. I want to avoid having mirrored traffic flowing between the hosts over the physical switch interfaces the hosts use.

What I think I want to do... Enable promiscuous mode on every vDPG/VLAN. Create a mirroring session that captures both tx/rx of all traffic for all VLANs on that particular host including traffic between VMs that also reside on the same host. Make the destination for that mirrored trafic with a VLAN that my sensor can attach to or make the destination the actual virtual interface of the VM.

Can anyone who has done this give the ELI5 version of how they did it? 

Thanks!

**Comment by charlespctech** [comment, marked as answer]
You would probably get more responses in a vmware specific discussion or subreddit
